vlingo-reactive-messaging-patterns/src/main/java/io/vlingo/reactive/messaging/patterns/messagefilter/InventorySystemMessageFilter.java

```diff
@@ -0,0 +1,32 @@
+// Copyright © 2012-2018 Vaughn Vernon. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the
+// Mozilla Public License, v. 2.0. If a copy of the MPL
+// was not distributed with this file, You can obtain
+// one at https://mozilla.org/MPL/2.0/.
+
+package io.vlingo.reactive.messaging.patterns.messagefilter;
+
+import io.vlingo.actors.Actor;
+import io.vlingo.actors.testkit.TestUntil;
+
+public class InventorySystemMessageFilter extends Actor implements InventorySystem {
+
+    private final TestUntil until;
+    private final InventorySystem inventorySystem;
+
+    public InventorySystemMessageFilter(final TestUntil until, final InventorySystem inventorySystem) {
+        this.inventorySystem = inventorySystem;
+        this.until = until;
+    }
+
+    @Override
+    public void orderPlaced(final OrderPlaced orderPlaced) {
+        if(orderPlaced.isType("TypeABC")) {
+            inventorySystem.orderPlaced(orderPlaced);
```

Comment:
Do you need `until.happened()` here?

Reply:
The idea is to ensure that the message was filtered in the test context. Is it an appropriate approach?